app/main/views/services.py

```diff
@@ -135,33 +134,21 @@ def update_service(service_id):
 
     json_payload = get_json_from_request()
     json_has_required_keys(json_payload,
-                           ["update_details", "services"])
+                           ['services', 'update_details'])
 
     update_json = json_payload['update_details']
     validate_updater_json_or_400(update_json)
-    service_update = drop_foreign_fields(
-        json_payload['services'],
-        ['supplierName', 'links', 'frameworkName', 'status']
-    )
-    json_has_matching_id(service_update, service_id)
 
-    data = dict(service.data.items())
-    data.update(service_update)
-    if "id" in data:
-        # It is an old-style service JSON with an id field
-        data["id"] = str(data["id"])
-    else:
-        # It is a new service JSON with id removed from payload already
-        data["id"] = service_id
+    json_has_matching_id(json_payload['services'], service_id)
 
-    detect_framework_or_400(data)
+    service.update_from_json(json_payload['services'],
+                             updated_by=update_json['updated_by'],
+                             updated_reason=update_json['update_reason'])
 
-    data = drop_foreign_fields(data, ['id'])
-    now = datetime.now()
-    service.data = data
-    service.updated_at = now
-    service.updated_by = update_json['updated_by']
-    service.updated_reason = update_json['update_reason']
+    data = service.serialize()
+    data = drop_foreign_fields(data,
+                               ['supplierName', 'links', 'frameworkName'])
```

Comment:
In the update it happens [here](https://github.com/alphagov/digitalmarketplace-api/pull/127/files#diff-413e9aa8961126d20e2d90befa13e0f1R231). I wanted to move the `from_json` work into the model so that it's closer to it's inverse (ie. `serialize`). It's not done for `import_service` as this was a bit trickier and it's documented as not being a model to follow.